src/client.coffee

```diff
@@ -1,163 +1,158 @@
-      log.trace('client connected')
-    )
-
-    @socket.on('data', (data) =>
-      @parser.execute(data)
-    )
-
-    @socket.on('end', ->
-      log.trace("Ensime server disconnected")
-    )
-
-    @socket.on('close', (data) ->
-      log.trace("Ensime server close event: " + data)
-    )
-
-    @socket.on('error', (data) ->
-      if (data.code == 'ECONNREFUSED')
-        log.error("Connection refused connecting to ensime, it is probably not running. Remove .ensime_cache/port and .ensime_cache/http and try again.")
-      else if (data.code == 'EADDRNOTAVAIL')
-        log.trace(data)
-        # happens when connecting too soon I think
-      else
-        log.trace("Ensime server error event: " + data)
-    )
-
-    @socket.on('timeout', ->
-      log.trace("Ensime server timeout event")
-    )
-
-  postString: (msg, callback) =>
-    swankMsg = Swank.buildMessage """{"req": #{msg}, "callId": #{@ensimeMessageCounter}}"""
-    @callbackMap[@ensimeMessageCounter++] = callback
-    log.trace("outgoing: " + swankMsg)
-    @socket.write(swankMsg, "UTF8")
-
-  # Public:
-  post: (msg, callback) ->
-    @postString(JSON.stringify(msg), callback)
-
-
-  getCompletions: (filePath, bufferText, offset, noOfAutocompleteSuggestions, callback) =>
-    tempFilePath = getTempDir() + filePath
-    fs.outputFile(tempFilePath, bufferText, (err) =>
-      if (err)
-        throw err
-      else
-        msg =
-          typehint: "CompletionsReq"
-          fileInfo:
-            file: filePath
-            contentsIn: tempFilePath
-          point: offset
-          maxResults: noOfAutocompleteSuggestions
-          caseSens: false
-          reload: true
-        @post(msg, callback)
-    )
-    
-    
- 
-  getSymbolAtPoint: (path, offset, callback) ->
-    req =
-      typehint: "SymbolAtPointReq"
-      file: path
-      point: offset
-    @post(req, (msg) ->
-      if msg.typehint == 'SymbolInfo'
-        callback(msg)
-      else
-        # if msg.typehint == 'FalseResponse'
-        # do nothing
-    )
     
 
-  typecheckBuffer: (path, text, callback = ->) =>
-    tempFilePath = getTempDir() + path
-    fs.outputFile(tempFilePath, text, (err) =>
-      if (err)
-        throw err
-      else
-        msg =
-          typehint: "TypecheckFileReq"
-          fileInfo:
-            file: path
-            contentsIn: tempFilePath
-        @post(msg, callback)
-    )
-
-
-  
-  typecheckFile: (path) =>
-    msg =
-      typehint: "TypecheckFileReq"
-      fileInfo:
-        file: path
-    @post(msg, (result) ->)
+    # Kills server if it was spawned from here.
+    destroy = ->
+      websocket.terminate()
+      serverPid?.kill()
 
-
-  symbolByName: (qualifiedName, callback) ->
-    msg =
-      typehint: 'SymbolByNameReq'
-      typeFullName: qualifiedName
-    @post(msg, callback)
     
+    postString = (msg, callback) ->
+      msg = """{"req": #{msg}, "callId": #{ensimeMessageCounter}}"""
+      callbackMap[ensimeMessageCounter++] = callback
```

Comment:
Hehe!

I actually really felt this was a dirty hack on an instinct level. There's some low instinct that I probably shouldn't make my own autoincrement id's, like in a sql db or whatever. Feels lika a lost opportunity too google and use a lib or something :)